### ISSUE TITLE
Incorrect behavior when translating an unsigned integer

### ISSUE THREAD

**Issue by de-odex** [PMunch/futhark#9]
This line is translated improperly:
```c
#define NC_NOBACKGROUND_MASK  0x8700000000000000ull
```
```nim
const
  Ncnobackgroundmask* = culonglong(-8718968878589280256'i64) ## Generated based on /home/deodex/Documents/CODE/Nim/natsuhi/notcurses/include/notcurses/notcurses.h:111:9
```

**Comment by PMunch** [comment]
Hmm, I guess Futhark should do `cast` instead of trying to convert the type. That value is the correct one, just represented as an int64 instead. If that line had been
```nim
const Ncnobackgroundmask* = cast[culonglong](-8718968878589280256'i64)
```
you will see that it has the correct value.

**Comment by de-odex** [comment]
Sorry, yes that is what I mean by translated improperly. I did not mean to imply the value is wrong, just that the conversion is invalid. Either a cast or interpreting unsigned and signed properly would work, the cast is easier.

**Comment by PMunch** [comment]
This should be fixed with the latest (0.4.0) version of Futhark
